### PR TITLE
Adding Xiaomi (Aqara) Smart Home Gateway #2

### PR DESCRIPTION
**Pull request by corbinmunce**
Is this better?  I see that Yeelight.cpp still has different spacing after I copied and pasted from the master file.  Is there recommended settings for Visual Studio that will help keep the formatting the same as the original?

### CODE REVIEW

**Comment by gizmocuz** [comment]
Our build system indicated that this branch still has some conflicts, see the message below (or above)
I think it is some kind of extension you are using in visual studio, but if you do not touch the yeelight file it would not have been changed
If you use tortoisegit, you can press the right mouse button on files and ask to revert the changes, probably this can also be done in visual studio directly

File: www/app/HardwareController.js [hunk 1]
```diff
@@ -1,72 +1,68 @@
 define(['app'], function (app) {
-    app.controller('HardwareController', [ '$scope', '$rootScope', '$location', '$http', '$interval', function($scope,$rootScope,$location,$http,$interval) {
+    app.controller('HardwareController', ['$scope', '$rootScope', '$location', '$http', '$interval', function ($scope, $rootScope, $location, $http, $interval) {
 
-        $scope.SerialPortStr=[];
+        $scope.SerialPortStr = [];
```
Comment (gizmocuz) [inline www/app/HardwareController.js]: Please undo all formatting changes.
Use notepad2-mod to edit if your editor keeps doing this

File: hardware/Yeelight.cpp [hunk 2]
```diff
@@ -283,12 +283,12 @@ bool Yeelight::WriteToHardware(const char *pdata, const unsigned char length)
...
-			ss << "{\"id\":1,\"method\":\"set_hsv\",\"params\":[" << cHue << ", 100, \"smooth\", 2000]}\r\n";
-			message = ss.str();
-		}
-		break;
+		sendOnFirst = true;
+		float cHue = (359.0f / 255.0f)*float(pLed->value); // hue given was in range of 0-255
+		ss << "{\"id\":1,\"method\":\"set_hsv\",\"params\":[" << cHue << ", 100, \"smooth\", 2000]}\r\n";
+		message = ss.str();
+	}
+								 break;
```
Comment (gizmocuz) [inline hardware/Yeelight.cpp]: Please do not submit changes for this file, as it is not related to this pull request

Comment (corbinmunce) [reply hardware/Yeelight.cpp]: Thanks for the tips, I'll be back a bit later with an updated commit

Comment (gizmocuz) [reply hardware/Yeelight.cpp]: Thanks!

**Comment by gizmocuz** [comment]
There seems to be still 2 files that have conflicts and causing the merge not to succeed

main/RFXNames.cpp
www/app/HardwareController.js

If you take them from the master repo, make your changes, and then commit, all should be fine

**Comment by corbinmunce** [comment]
All I'm doing is adding this line to RFXNames.cpp:
		{ HTYPE_XiaomiGateway, "Xiaomi Gateway" },

and it then fails the check.

**Comment by corbinmunce** [comment]
Just missing an addition to line 244 in RFXNames.cpp:

{ HTYPE_XiaomiGateway, "Xiaomi Gateway" },

**Comment by corbinmunce** [comment]
I didn't modify RFXNames.cpp and HardwareController.js, and the checks passed ok.  I used notepad2-mod to edit the files with the Xiaomi additions, and the check have failed again.

**Comment by corbinmunce** [comment]
Could it be this error message:
"Only those with write access to this repository can merge pull requests."

Due to me not having permissions to modify RFXNames.cpp?

**Comment by gizmocuz** [comment]
Nope... All is fine now
Don you want me to merge this?

**Comment by corbinmunce** [comment]
yes please merge, all the files I needed have now passed!

or not... had taken out the Xiaomi line from RFXNames.cpp

**Comment by gizmocuz** [comment]
Thanks!

**Comment by corbinmunce** [comment]
Still had that problem with RFXNames.cpp, it needs this added at line 244:
`{ HTYPE_XiaomiGateway, "Xiaomi Gateway" },`

Can you add this?  Otherwise it I'll try again in my next pull request.

**Comment by gizmocuz** [comment]
Please fork again and make a new pull request

**Pull request merged**